app/app/models/pinwheel_account.rb

```diff
@@ -29,11 +33,14 @@ def job_succeeded?(job)
     supported_jobs.include?(job) && send(sync_column).present? && send(error_column).blank?
   end
 
-  def job_completed?(job)
+  def synchronization_status?(job)
     error_column, sync_column = event_columns_for(job)
     return nil unless error_column.present?
 
-    supported_jobs.include?(job) && (send(sync_column).present? || send(error_column).present?)
+    return :unsupported unless supported_jobs.include?(job)
+    return :succeeded if job_succeeded?(job)
+    return :in_progress if supported_jobs.include?(job) && (send(sync_column).blank? && send(error_column).blank?)
+    :failed if supported_jobs.include?(job) && (send(error_column).present?)
```

Comment:
I found it a bit hard to follow this logic - despite looking like guard clauses, these conditionals are all coupled together so they don't really allow a reader to ignore them as preconditions of the actual method's logic (in the way that a guard clause would).

I might suggest just using the old if/else constructs....

```suggestion
    if supported_jobs.exclude?(job)
      :unsupported
    elsif job_succeeded?(job)
      :succeeded
    elsif supported_jobs.include?(job) && (send(sync_column).blank? && send(error_column).blank?)
      :in_progress
    elsif supported_jobs.include?(job) && (send(error_column).present?)
      :failed
    end
```

This simpler code also makes it apparent that we could also simplify the last two conditions by removing `supported_jobs.include?(job)` since it will always be true if we've reached that branch (as the first branch will be executed instead if `supported_jobs.exclude?(job)`)